### ISSUE TITLE
Exit with failure when the list of observations is not empty

### ISSUE THREAD

**Issue by chshersh** [kowainik/stan#44]
So the tool can be easily used on CI

**Comment by chshersh** [comment]
It's a one-line issue with the current implementation of `Analysis` data type. But if we move `Observations` inside `FileInfo`, as we discussed before, we won't be able to check if we any observations easily. We will need to check whether at least one value inside `FileMap` has non-empty list of observation.

**Comment by chshersh** [comment]
As discussed with @vrom911, it would be ideal to control via CLI when to fail, for example, only when the severity of resulting observations is `Error` or higher.

**Comment by chshersh** [comment]
A shower thought: how much flexibility do we want to give to this failure mode? Do we want to duplicate some logic from the `check` data type and allow users to define custom failure rules, e.g.

1. Fail if any observation has severity `Warning` or higher.
2. Fail on particular inspection `STAN-0206`.
3. Fail on any inspection in this file (which is supposed to be super-safe).

In the TOML config this can look like this:

```toml
[[fail]]
severity = "Warning"
scope = "all"

[[fail]]
id = "STAN-0206"
scope = "all"

[[fail]]
filter = "all"
file = "src/Super/Safe/API.hs"
```

**Comment by chshersh** [comment]
For now, we decided to implement failure only if at least one `Observation` has severity `Error`. Most customizable failure settings will be implemented during work on #246.
